core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -143,6 +151,8 @@ private UnsafeExternalSorter(
       this.inMemSorter = existingInMemorySorter;
     }
     this.peakMemoryUsedBytes = getMemoryUsage();
+    this.numElementsForSpillThreshold =
+      SparkEnv.get().conf().getLong("spark.shuffle.spill.numElementsForceSpillThreshold", (1024 * 1024 * 1024 / 8));
```

Comment:
it should be 1024 * 1024 * 1024 / 2 right? Its useful for us to have this configurable, because we have been seeing JVM OOMs and SEG faults when we are allocating ridiculously large buffer in memory.